# 多相机拼接和目标跟踪接口的配置文件
# 配置说明：请根据实际设备参数修改以下配置

# rtsp_image_2 / camera_image_topic_0 !!!!!!!!!!!!


# ============ 参数配置 ============

# 相机发布器参数配置
CameraPublisher_parameters:
  input_frame_rate: 25             # 相机设置的输入帧率
    
parameters:
  # 主程序的参数配置
  Main_parameters:
    # 订阅的相机配置列表（支持动态增减相机数量）
    cameras:
      - camera_name: "camera_0"           # 相机名称（用于映射）
        topic_name: "/rtsp_image_0"   # 相机0的话题名 rtsp_image_2 / camera_image_topic_0
        camera_index: 0                       # 相机索引
      - camera_name: "camera_1"           # 相机名称（用于映射）
        topic_name: "/rtsp_image_1"   # 相机1的话题名
        camera_index: 1                       # 相机索引
      - camera_name: "camera_2"           # 相机名称（用于映射）
        topic_name: "/rtsp_image_2"   # 相机2的话题名 
        camera_index: 2                       # 相机索引
    # 如需添加更多相机，按照上面的格式继续添加即可
    
    JHjpgMsg_topic: "/image_topic_all"         # JHjpgMsg话题名
    publish_timeout: 5                       # 发布超时时间，单位秒
    fus_trajectory_topic: "/fus_trajectory_topic" # 船只跟踪结果的话题名
    gnss_topic: "/gnss_topic"          # GNSS话题名
    get_camera_params_service: "/get_camera_params_service" # 获取相机参数服务的话题名
    
  # 对应的拼接器的参数配置
  Stitcher_parameters:
    refresh_time: 5                 # 拼缝线刷新时间间隔，单位秒（3600秒=1小时/86400秒=24小时，如果场景稳定可以设置更长）
    min_key_points: 100               # 特征匹配最小关键点数
    min_confidence: 0.25              # 匹配对最小置信度(0.5->0.3)
    min_inliers: 4                  # 特征匹配最小内点数(10->5)
    max_focal_variance: 50000.0      # 各个相机之间最大焦距方差
    y_tolerance: 200.0               # 拼接图像间的y轴容差
    roi_threshold: 0.95              # 掩码去除黑边的ROI阈值
    detect_confidence: 0.05          # 目标检测置信度
    iou_threshold: 0.1               # 目标检测框IOU筛选阈值
    scale: 0.6                       # 图像缩放比例
    crop_or_not: true                  # 是否裁剪全景图 
    crop_top: 230                         # 裁剪上边界像素
    crop_bottom: 170                      # 裁剪下边界像素
    crop_left: 50                        # 裁剪左边界像素
    crop_right: 50                       # 裁剪右边界像素
    draw_box_or_not: true               # 是否绘制检测框
    save_camera_params: false         # 是否保存相机参数
    remap_dir: "install/image_stitching_pkg/share/image_stitching_pkg/config/remap"   # 重映射文件保存路径
    cache_file: "install/image_stitching_pkg/share/image_stitching_pkg/config/remap/stitch_cache.yaml" # 其余缓存文件路径
    calib_dir: "install/image_stitching_pkg/share/image_stitching_pkg/config"   # 固定的相机标定内参文件路径
    use_saved_camera_params: false     # 是否使用已保存的相机K R T参数
    FOV_hor: 105.0                   # 水平视场角
    FOV_ver: 57.0                    # 垂直视场角
